flag.go

```diff
@@ -881,7 +887,29 @@ func (f *FlagSet) parseArgs(args []string) (err error) {
 func (f *FlagSet) Parse(arguments []string) error {
 	f.parsed = true
 	f.args = make([]string, 0, len(arguments))
-	err := f.parseArgs(arguments)
+	err := f.parseArgs(arguments, nil)
```

Comment:
What do you think about defining a function here which does the:
```
err = f.setFlag(flag, value, shorthands)
```
So that `ParseAll()` when called with a custom function will have no implied side effects.

It also means we wouldn't ever have to test for fn == nil.